hydrolib/core/basemodel.py

```diff
@@ -808,13 +886,13 @@ def __init__(
 
             logger.info(f"Loading data from {filepath}")
 
-            data = self._load(loading_path)
-            context.register_model(filepath, self)
-            data["filepath"] = filepath
+            data = context.retrieve_model(filepath)
+            if context.is_content_changed(filepath):
+                data = self._load(loading_path)
+                context.register_model(filepath, self)
+                data["filepath"] = filepath
```

Comment:
Shouldn't the filepath be set regardless of whether the data has been cached or not? Another question: If we find data in the cache, do we still need to do all the other steps below? If a file is referenced multiple times, do we need separate instances of models, or should they refer to the same instance?

If `context.retrieve_model(filepath)` returns something (i.e. the filepath is in the cache), `data` is a `FileModel`. If the model is not cached, `data` will be a dictionary.

I think we should discuss this.

Reply:
we worked this out together.
I now also added rereading when file has changed back in, in the new.